### ISSUE TITLE
OneAgent SDK causing breaking changes with DB cursors

### ISSUE THREAD

**Issue by alon-selfinc** [dynatrace-oss/OneAgent-SDK-Python-AutoInstrumentation#76]
Hi! We are trying to incorporate the OneAgent SDK into our codebase.
Currently on `oneagent-sdk~=1.5.0.20221221.150433`, `psycopg2==2.9.5` running on Python 3.7.2. 

Our `create_app()` flask function essentially runs this:

```
import oneagent

init_result = oneagent.initialize()
sdk = oneagent.get_sdk()
```

After incorporating the SDK, all of our DAO objects/queries are behaving differently.
We can no longer pull a field value by name out of a DB cursor:

```
odfi_uuid = None if dfi is None else dfi['uuid']
E       TypeError: tuple indices must be integers or slices, not str
```

We'd really appreciate any suggestions on tackling the issue. Please let me know if I can further clarify anything here. Thanks!

**Comment by dlopes7** [comment]
I am having a hard time reproducing, would you mind sharing a piece of code that allows you to access the rows like that?

When I run

```python
cur = conn.cursor()
cur.execute("SELECT tablename, tableowner FROM pg_catalog.pg_tables")
for row in cur:
    print(row['tablename'])
```

I get the same error as you, even without autodynatrace installed

**Comment by dlopes7** [comment]
Are you per chance using 

```python
cursor_factory=RealDictCursor
```

In the connection creation? I think I was able to reproduce by using that, and will look to implement a fix

**Comment by alon-selfinc** [comment]
@dlopes7 you are correct! we are using `DictCursor`. sorry I should have clarified in my initial post. thanks so much for investigating!

**Comment by caseycoding** [comment]
Hi @dlopes7!  There might be a minor misspelling in [this update]( https://github.com/dynatrace-oss/OneAgent-SDK-Python-AutoInstrumentation/commit/49091a048dac1c550399fed4460e81034e5a384c.):  [extras](https://github.com/psycopg/psycopg2/blob/master/lib/extras.py) vs `extra`.  Is it ok if we make MRs for you to review going forward?  Don't want to toss it all your way.   Thanks!

**Comment by davetonkin-LC** [comment]
@dlopes7 - Hi - just checking to see if you saw Casey's last comment.  Are you able to reopen this and/or create a new issue to address the misspelling that he mentioned?
